src/dra/843B0.c

```diff
@@ -2303,7 +2303,225 @@ void EntityAguneaHitEnemy(Entity* self) {
+                break;
+            } else {
+                a2 = s2->ext.et_80129864.ent->posX.i.hi;
+                v1 = s2->ext.et_80129864.ent->posY.i.hi;
+            }
+        } else {
+            if (s2->facingLeft) {
+                a2 = 0x140;
+                v1 = 0xA0;
+            } else {
+                a2 = -0x40;
+                v1 = 0xA0;
+            }
+            if (s2->params & 1) {
+                v1 += 0x40;
+            }
+        }
+
+        s1 = ratan2(s2->posY.i.hi - v1, a2 - s2->posX.i.hi) & 0xFFF;
+        s3 = s2->ext.et_80129864.unk80 & 0xFFF;
+        v1 = abs(s3 - s1);
+        angle_offset = CLAMP_MAX(angle_offset, v1);
+        if (s3 < s1) {
+            if (v1 < 0x800) {
+                s3 += angle_offset;
+            } else {
+                s3 -= angle_offset;
+            }
+        } else {
+            if (v1 < 0x800) {
+                s3 -= angle_offset;
+            } else {
+                s3 += angle_offset;
+            }
+        }
+        s2->ext.et_80129864.unk80 = s3 & 0xFFF;
+        if (++s2->ext.et_80129864.unk84 > 0x60) {
+            s2->step++;
+        }
+        break;
+    case 3:
+        s2->flags = 0x08910000;
+        s1 = rcos(s2->ext.et_80129864.unk82) >> 8;
+        s2->ext.et_80129864.unk82 += 0x80;
+        s2->ext.et_80129864.unk80 += s1;
+        break;
+    }
+
+    s2->velocityX = rcos(s2->ext.et_80129864.unk80) << 6;
+    s2->velocityY = -(rsin(s2->ext.et_80129864.unk80) << 6);
+    s2->posX.val += s2->velocityX;
+    s2->posY.val += s2->velocityY;
+
+    if (s2->facingLeft) {
+        s2->rotZ = s2->ext.et_80129864.unk80;
+    } else {
+        s2->rotZ = 0x800 - s2->ext.et_80129864.unk80;
+    }
+
+    s0 = &g_PrimBuf[s2->primIndex];
+    for (s5 = 0; s5 < 8; s5++) {
+        if (s2->ext.et_80129864.unk86 == s5) {
+            s0->x1 = 1;
+        }
+
+        switch (s0->x1) {
+        case 0:
+            s8 = 0;
+            break;
+        case 1:
+            s8 = 1;
+            s0->x1 += 1;
+            s0->y1 = 0x100;
+            s0->x0 = s2->posX.i.hi;
+            s0->y0 = s2->posY.i.hi;
+            s0->x2 = s2->ext.et_80129864.unk80;
+            break;
+        case 2:
+            s0->y1 -= 16;
+            s8 = 2;
+            break;
+        }
+
+        s3 = s0->x2;
+        s6 = s0->x0;
+        s7 = s0->y0;
+        s4 = s0->y1;
+        s0 = s0->next;
+
+        switch (s8) {
+        case 0:
+            s0->drawMode |= 8;
+            break;
+        case 1:
+            temp_u = D_800B0846[s2->animCurFrame * 2];
+            temp_v = D_800B0846[s2->animCurFrame * 2 + 1];
+            s0->u0 = s0->u2 = temp_u;
+            s0->u1 = s0->u3 = temp_u + 31;
+            s0->v0 = s0->v1 = temp_v;
+            s0->v2 = s0->v3 = temp_v + 15;
+            s0->r0 = s0->b0 = s0->g0 = s0->r1 = s0->b1 = s0->g1 = s0->r2 =
+                s0->b2 = s0->g2 = s0->r3 = s0->b3 = s0->g3 = 0x80;
+            s0->drawMode &= 0xFFF7;
```

Comment:
```suggestion
            s0->drawMode &= ~DRAW_HIDE;
```
this and a few other `drawMode` you can document